analysis_runner/git.py

```diff
@@ -129,6 +141,59 @@ def check_if_commit_is_on_remote(commit: str) -> bool:
         return False
 
 
+def guess_script_name_from_script_argument(script: list[str]) -> Optional[str]:
+    """
+    Guess the script name from the first argument of the script.
+    If the first argument is an executable, try the second param
+
+    >>> guess_script_name_from_script_argument(['python', 'main.py'])
+    'main.py'
+
+    >>> guess_script_name_from_script_argument(['./main.sh'])
+    'main.sh'
+
+    >>> guess_script_name_from_script_argument(['main.sh'])
+    'main.sh'
+
+    >>> guess_script_name_from_script_argument(['./test/path/main.sh', 'arg1', 'arg2'])
+    'test/path/main.sh'
+
+    >>> guess_script_name_from_script_argument(['gcloud', 'cp' 'test'])
+    None
+
+    """
+    executables = {'python', 'python3', 'bash', 'sh'}
```

Comment:
RScript might be a solid addition to this list?

Reply:
Ah it was in the chatgpt suggestion, I think I removed it instead of python 🤦